packages/react-dom/src/server/ReactPartialRenderer.js

```diff
@@ -945,7 +946,8 @@ class ReactDOMServerRenderer {
       }
       if (typeof elementType === 'object' && elementType !== null) {
         switch (elementType.$$typeof) {
-          case REACT_FORWARD_REF_TYPE: {
+          case REACT_FORWARD_REF_TYPE:
+          case REACT_PURE_TYPE: {
             const element: ReactElement = ((nextChild: any): ReactElement);
             const nextChildren = toArray(
               elementType.render(element.props, element.ref),
```

Comment:
This is not correct. It is passing the `element.ref` second argument. It shouldn't get the second argument in the `pure` case.

Reply:
But https://github.com/facebook/react/pull/13822?

Reply:
I didn't see that one. I'm not sure I approve. Let me think about it.